### PR TITLE
fix(cmd): lock repository file during `helm repo add`

### PR DESCRIPTION
**Pull request by bacongobbler**
port of #5678

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: cmd/helm/repo_add_test.go [hunk 1]
```diff
@@ -86,3 +88,51 @@ func TestRepoAdd(t *testing.T) {
...
+	}
+
+	var name string
+	for i := 0; i < 3; i++ {
+		name = fmt.Sprintf("%s-%d", testName, i)
+		if !f.Has(name) {
+			t.Errorf("%s was not successfully inserted into %s: %s", name, repoFile, f.Repositories[0])
+		}
+	}
+}
```
Comment (thomastaylor312) [inline cmd/helm/repo_add_test.go]: Did we want to pull over the [subprocess test](https://github.com/helm/helm/pull/5678/files#diff-5537931905cc99a35fc6d296d4e3feadR154) as well? I figure this concurrency locking will most likely occur with multiple processes

**Review by thomastaylor312 — COMMENTED** [review]
One discussion item before LGTM

File: cmd/helm/repo_add_test.go (hunk 1 above)
Comment (bacongobbler) [reply cmd/helm/repo_add_test.go]: The subprocess test may likely be the cause of the circleci failures in helm 2. It's excessive, flaky, and unnecessary IMO. The parallel waitgroup test should be more than enough to cover the case

**Comment by mattfarina** [comment]
@bacongobbler the failed CI is for a linting oops. Should be simple to fix here.

File: cmd/helm/repo_add_test.go (hunk 1 above)
Comment (thomastaylor312) [reply cmd/helm/repo_add_test.go]: 👌

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**